src/applications/model/gaming-traffic-model.cc

```diff
@@ -0,0 +1,104 @@
+/* -*- Mode:C++; c-file-style:"gnu"; indent-tabs-mode:nil; -*- */
+/*
+ * Copyright (c) 2020, University of Padova, Department of Information
+ * Engineering, SIGNET Lab.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation;
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ * Authors: Salman Mohebi <s.mohebi22@gmail.com>
+ *
+ */
+
+#include "ns3/log.h"
+#include "ns3/simulator.h"
+#include "gaming-traffic-model.h"
+
+namespace ns3 {
+
+  NS_LOG_COMPONENT_DEFINE ("GamingTrafficModel");
+
+  NS_OBJECT_ENSURE_REGISTERED(GamingTrafficModel);
+
+  TypeId
+  GamingTrafficModel::GetTypeId(void) {
+    static TypeId tid = TypeId("ns3::GamingTrafficModel")
+        .SetParent<RandomVariableStream>()
+        .SetGroupName("GamingTraffic")
+        .AddConstructor<GamingTrafficModel>()
+    ;
+    return tid;
+  }
+
+  GamingTrafficModel::GamingTrafficModel() {
+    NS_LOG_FUNCTION (this);
+  }
+
+  double
+  GamingTrafficModel::GetValue(void) {
+    NS_LOG_FUNCTION (this);
+    if (IsValid()) {
+      double thresh = CreateObjectWithAttributes<UniformRandomVariable>()->GetValue();
+      double cumProb = 0;
+      for (int index = 0; index < (int) m_prob.size(); ++index) {
+        cumProb += m_prob[index];
+        if (thresh <= cumProb) {
+          return m_rng[index]->GetValue();
+        }
+      }
+      return m_rng.back()->GetValue();
+    }
+  else
+  {
+    NS_ASSERT_MSG (false, "Sum of the m_prob should be one.");
+  }
+  }
+
+  uint32_t
+  GamingTrafficModel::GetInteger (void)
+  {
+    NS_LOG_FUNCTION (this);
+    return (uint32_t)GetValue ();
+  }
+
+  void
+  GamingTrafficModel::SetProb (std::vector < double > prob)
+  {
+    NS_LOG_FUNCTION (this);
+    m_prob = prob;
+  }
+
+  void
+  GamingTrafficModel::SetRngs (std::vector <Ptr<RandomVariableStream> > rng)
+  {
+    NS_LOG_FUNCTION (this);
+    m_rng = rng;
+  }
```

Comment:
Maybe it's better to have a `AddRandomVarible` or sth, which receives an RV with its probability and adds it to the vectors. (In the case we want to add RVs one by one)
Another one, `SetRandomVariables` which set the set RVs ones for all, instead of having two different methods. (as you said)

Reply:
i would avoid creating the method `AddRandomVariable` since it's not needed for the moment and it might add some nasty bugs